test/project_types/node/commands/create_test.rb

```diff
@@ -30,20 +30,37 @@ def test_prints_help_with_no_name_argument
       end
 
       def test_check_node_installed
-        @context.expects(:capture2e).with('node', '-v').returns([nil, mock(success?: false)])
+        @context.expects(:capture2e).with('which', 'node').returns([nil, mock(success?: false)])
         assert_raises ShopifyCli::Abort, Create::NODE_REQUIRED_NOTICE do
           perform_command
         end
       end
 
+      def test_check_get_node_version
+        @context.expects(:capture2e).with('which', 'node').returns(['/usr/bin/node', mock(success?: true)])
+        @context.expects(:capture2e).with('node', '-v').returns([nil, mock(success?: false)])
+        assert_raises ShopifyCli::Abort, Create::NODE_VERSION_FAILURE_NOTICE do
+          perform_command
+        end
+      end
+
       def test_check_npm_installed
         Create.any_instance.stubs(:check_node)
-        @context.expects(:capture2e).with('npm', '-v').returns([nil, mock(success?: false)])
+        @context.expects(:capture2e).with('which', 'npm').returns([nil, mock(success?: false)])
         assert_raises ShopifyCli::Abort, Create::NPM_REQUIRED_NOTICE do
           perform_command
         end
       end
 
+      def test_check_get_npm_version
+        Create.any_instance.stubs(:check_node)
```

Comment:
I see we're already doing it above, and it's not critical, but in general I think we should try to avoid stubbing on the class being tested:

https://thoughtbot.com/blog/don-t-stub-the-system-under-test

Reply:
This stubbing is quite easy to get rid of by adding mocks for the node check commands, so I agree that we can just drop it.